openupgrade_scripts/scripts/website_sale_stock/15.0.1.0/post-migration.py

```diff
@@ -0,0 +1,23 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE product_template
+            SET allow_out_of_stock_order = CASE
+            WHEN inventory_availability = 'never' THEN true ELSE false
+            END
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE product_template
+            SET show_availability = CASE
+            WHEN inventory_availability = 'threshold' THEN true ELSE false
+            END
```

Comment:
```suggestion
            UPDATE product_template
        SET allow_out_of_stock_order = CASE
            WHEN inventory_availability IN ('never', 'custom') THEN true ELSE false END,
        show_availability = CASE
            WHEN inventory_availability IN ('always', 'threshold') THEN true ELSE false END,
        available_threshold = CASE WHEN inventory_availability = 'always' THEN 999999999 END
```

You can do it in one query and coverage all options from older versions